presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -697,13 +724,29 @@ private StreamProperties(
             this.ordered = ordered;
             checkArgument(!ordered || distribution == SINGLE, "Ordered must be a single stream");
 
-            this.otherActualProperties = otherActualProperties;
+            this.otherActualProperties = requireNonNull(otherActualProperties);
+            requireNonNull(streamPropertiesFromUniqueColumn).ifPresent(properties -> checkArgument(!properties.streamPropertiesFromUniqueColumn.isPresent()));
+            this.streamPropertiesFromUniqueColumn = streamPropertiesFromUniqueColumn;
         }
 
         public List<LocalProperty<VariableReferenceExpression>> getLocalProperties()
         {
-            checkState(otherActualProperties != null, "otherActualProperties not set");
-            return otherActualProperties.getLocalProperties();
+            checkState(otherActualProperties.isPresent(), "otherActualProperties not set");
+            return otherActualProperties.get().getLocalProperties();
+        }
+
+        public List<LocalProperty<VariableReferenceExpression>> getAdditionalLocalProperties()
+        {
+            checkState(otherActualProperties.isPresent(), "otherActualProperties not set");
```

Comment:
I see it's basically just copying the previous behavior of getLocalProperties, but I'm curious why these two methods throw rather than return an empty list.